pallets/summary/src/lib.rs

```diff
@@ -728,6 +752,55 @@ pub mod pallet {
         }
     }
     impl<T: Config<I>, I: 'static> Pallet<T, I> {
+        pub fn set_summary_status(
+            root_id: RootId<BlockNumberFor<T>>,
+            status: SummaryStatus,
+        ) -> DispatchResult {
+            <RootStatus<T, I>>::insert(
+                root_id.range,
+                root_id.ingress_counter,
+                status
+            );
+            Ok(())
+        }
+
+        pub fn process_accepted_summary(
+            root_id: &RootId<BlockNumberFor<T>>,
+        ) -> DispatchResult {
+            let root_data = Self::try_get_root_data(root_id)?;
+            if root_data.root_hash != Self::empty_root() {
+                if T::AutoSubmitSummaries::get() {
+                    Self::send_root_to_ethereum(root_id, &root_data)?;
+                } else {
+                    let approved_root_id = Self::get_next_approved_root_id()?;
+                    <AnchorRoots<T, I>>::insert(approved_root_id, root_data.root_hash);
+                }
+            }
+            Ok(())
+        }
+
+        pub fn set_summary_status_and_process(
+            root_id: RootId<BlockNumberFor<T>>,
+            status: SummaryStatus,
+        ) -> DispatchResult {
+            Self::set_summary_status(root_id, status.clone())?;
+            
+            if status == SummaryStatus::Accepted {
+                Self::process_accepted_summary(&root_id)?;
+            }
+            Ok(())
+        }
+
+        pub fn update_status_if_required(
+            root_id: RootId<BlockNumberFor<T>>,
+            status: SummaryStatus,
+        ) -> DispatchResult {
+            if T::RequireWatchtowerValidation::get() {
```

Comment:
nit: We duplicate the check here and in the caller. Maybe have the caller check and remove the `RequireWatchtowerValidation` from here? or vise versa?